'flutter build linux' fails with "Build process failed"

<!-- Thank you for using Flutter!

     If you are looking for support, please check out our documentation
     or consider asking a question on Stack Overflow:
      * https://flutter.dev/
      * https://api.flutter.dev/
      * https://stackoverflow.com/questions/tagged/flutter?sort=frequent

     If you have found a bug or if our documentation doesn't have an answer
     to what you're looking for, then fill our the template below. Please read
     our guide to filing a bug first: https://flutter.dev/docs/resources/bug-reports
-->

## Steps to Reproduce

`flutter create testapp` and `flutter build -v linux` inside that folder.  `flutter run -d linux` works without issue.

```
$ flutter run -d linux
Launching lib/main.dart on Linux in debug mode...
Building Linux application...                                           
Waiting for Linux to report its views...                             4ms
Syncing files to device Linux...                                   129ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application
running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on Linux is available
at: http://127.0.0.1:42977/wG0TJPc_XE4=/
Lost connection to device.
➜  lindesk flutter build -v linux
[   +8 ms] executing: [/home/mark/development/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +23 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] eb1a6efe1637b6213d94ad6e2d159a3d50e56699
[        ] executing: [/home/mark/development/flutter/] git tag --contains HEAD
[ +157 ms] Exit code 0 from: git tag --contains HEAD
[   +2 ms] executing: [/home/mark/development/flutter/] git describe --match *.*.*-*.*.pre --first-parent --long
--tags
[ +127 ms] Exit code 0 from: git describe --match *.*.*-*.*.pre --first-parent --long --tags
[        ] 1.19.0-1.0.pre-213-geb1a6efe16
[   +6 ms] executing: [/home/mark/development/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] executing: [/home/mark/development/flutter/] git ls-remote --get-url origin
[   +4 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +47 ms] executing: [/home/mark/development/flutter/] git rev-parse --abbrev-ref HEAD
[   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[  +37 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +11 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +9 ms] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[ +218 ms] Generating /home/mark/AndroidStudioProjects/lindesk/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +33 ms] Building Linux application...
[   +6 ms] executing: [build/linux/release/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Release /home/mark/AndroidStudioProjects/lindesk/linux
[  +26 ms] -- Configuring done
[        ] -- Generating done
[   +7 ms] -- Build files have been written to: /home/mark/AndroidStudioProjects/lindesk/build/linux/release
[   +7 ms] executing: ninja -C build/linux/release install
[   +5 ms] ninja: Entering directory `build/linux/release'
[+5680 ms] [1/15] Generating /home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/libflutter_linux_glfw.so, flutter_export.h, flutter_glfw.h, flutter_messenger.h, flutter_plugin_registrar.h,
/home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/engine_method_result.cc, /home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/standard_codec.cc,
/home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/plugin_registrar.cc, /home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/flutter_engine.cc,
/home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/flutter_window_controller.cc, _phony_
[   +2 ms] FAILED: /home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/libflutter_linux_glfw.so flutter/flutter_export.h flutter/flutter_glfw.h flutter/flutter_messenger.h flutter/flutter_plugin_registrar.h
/home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/engine_method_result.cc /home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/standard_codec.cc
/home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/plugin_registrar.cc /home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/flutter_engine.cc
/home/mark/AndroidStudioProjects/lindesk/linux/flutter/ephemeral/cpp_client_wrapper_glfw/flutter_window_controller.cc flutter/_phony_
[   +1 ms] cd /home/mark/AndroidStudioProjects/lindesk/build/linux/release/flutter && /usr/bin/cmake -E env FLUTTER_ROOT="/home/mark/development/flutter" PROJECT_DIR="/home/mark/AndroidStudioProjects/lindesk" DART_OBFUSCATION="false"
TRACK_WIDGET_CREATION="true" TREE_SHAKE_ICONS="true" FLUTTER_TARGET="lib/main.dart" /home/mark/development/flutter/packages/flutter_tools/bin/tool_backend.sh linux-x64 Release
[        ] [   +8 ms] executing: [/home/mark/development/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [  +24 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [        ] eb1a6efe1637b6213d94ad6e2d159a3d50e56699
[        ] [        ] executing: [/home/mark/development/flutter/] git tag --contains HEAD
[        ] [ +148 ms] Exit code 0 from: git tag --contains HEAD
[        ] [   +2 ms] executing: [/home/mark/development/flutter/] git describe --match *.*.*-*.*.pre --first-parent --long --tags
[        ] [ +125 ms] Exit code 0 from: git describe --match *.*.*-*.*.pre --first-parent --long --tags
[        ] [        ] 1.19.0-1.0.pre-213-geb1a6efe16
[        ] [   +6 ms] executing: [/home/mark/development/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[        ] [   +4 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] [        ] origin/master
[        ] [        ] executing: [/home/mark/development/flutter/] git ls-remote --get-url origin
[        ] [   +3 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] [        ] https://github.com/flutter/flutter.git
[        ] [  +42 ms] executing: [/home/mark/development/flutter/] git rev-parse --abbrev-ref HEAD
[        ] [   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] master
[        ] [  +32 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [   +5 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] [  +75 ms] Initializing file store
[        ] [  +10 ms] Done initializing file store
[        ] [  +38 ms] Skipping target: gen_localizations
[        ] [   +1 ms] unpack_linux: Starting due to {}
[        ] [ +485 ms] Skipping target: kernel_snapshot
[        ] [   +3 ms] aot_elf_release: Starting due to {}
[        ] [   +4 ms] executing: /home/mark/development/flutter/bin/cache/artifacts/engine/linux-x64-release/gen_snapshot --deterministic --snapshot_kind=app-aot-elf
--elf=/home/mark/AndroidStudioProjects/lindesk/.dart_tool/flutter_build/e66215950b7040a0d8e3dcf11894e858/app.so --strip --no-causal-async-stacks --lazy-async-stacks
/home/mark/AndroidStudioProjects/lindesk/.dart_tool/flutter_build/e66215950b7040a0d8e3dcf11894e858/app.dill
[        ] [   +1 ms] Persisting file store
[        ] [   +7 ms] Done persisting file store
[        ] [   +7 ms] "flutter assemble" took 652ms.
[        ] Oops; flutter has exited unexpectedly: "Invalid argument(s): Cannot find executable for /home/mark/development/flutter/bin/cache/artifacts/engine/linux-x64-release/gen_snapshot.".
[        ] [+1560 ms] executing: /opt/android-studio/jre/bin/java -version
[        ] [  +41 ms] Exit code 0 from: /opt/android-studio/jre/bin/java -version
[        ] [        ] openjdk version "1.8.0_202-release"
[        ]            OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
[        ]            OpenJDK 64-Bit Server VM (build 25.202-b49-5587405, mixed mode)
[        ] [  +15 ms] executing: /opt/android-studio/jre/bin/java -version
[        ] [  +44 ms] Exit code 0 from: /opt/android-studio/jre/bin/java -version
[        ] [        ] openjdk version "1.8.0_202-release"
[        ]            OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
[        ]            OpenJDK 64-Bit Server VM (build 25.202-b49-5587405, mixed mode)
[        ] [   +5 ms] executing: [/home/mark/development/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%ar
[        ] [   +4 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%ar
[        ] [        ] 10 hours ago
[        ] [   +7 ms] executing: /home/mark/development/flutter/bin/cache/artifacts/engine/android-arm-profile/linux-x64/gen_snapshot
[        ] [   +4 ms] Exit code 255 from: /home/mark/development/flutter/bin/cache/artifacts/engine/android-arm-profile/linux-x64/gen_snapshot
[        ] [        ] At least one input is required
[        ]            Usage: gen_snapshot [<vm-flags>] [<options>] <dart-kernel-file>             
[        ]            Common options:                                                             
[        ]            --help                                                                      
[        ]              Display this message (add --verbose for information about all VM options).
[        ]            --version                                                                   
[        ]              Print the VM version.                                                     
[        ]            To create a core snapshot:                                                  
[        ]            --snapshot_kind=core                                                        
[        ]            --vm_snapshot_data=<output-file>                                            
[        ]            --isolate_snapshot_data=<output-file>                                       
[        ]            <dart-kernel-file>                                                          
[        ]            To create an AOT application snapshot as assembly suitable for compilation  
[        ]            as a static or dynamic library:                                             
[        ]            --snapshot_kind=app-aot-assembly                                            
[        ]            --assembly=<output-file>                                                    
[        ]            [--strip]                                                                   
[        ]            [--obfuscate]                                                               
[        ]            [--save-debugging-info=<debug-filename>]                                    
[        ]            [--save-obfuscation-map=<map-filename>]                                     
[        ]            <dart-kernel-file>                                                          
[        ]            To create an AOT application snapshot as an ELF shared library:             
[        ]            --snapshot_kind=app-aot-elf                                                 
[        ]            --elf=<output-file>                                                         
[        ]            [--strip]                                                                   
[        ]            [--obfuscate]                                                               
[        ]            [--save-debugging-info=<debug-filename>]                                    
[        ]            [--save-obfuscation-map=<map-filename>]                                     
[        ]            <dart-kernel-file>                                                          
[        ]            AOT snapshots can be obfuscated: that is all identifiers will be renamed    
[        ]            during compilation. This mode is enabled with --obfuscate flag. Mapping     
[        ]            between original and obfuscated names can be serialized as a JSON array     
[        ]            using --save-obfuscation-map=<filename> option. See dartbug.com/30524       
[        ]            for implementation details and limitations of the obfuscation pass.
[        ] [  +93 ms] executing: /home/mark/Android/Sdk/platform-tools/adb devices -l
[        ] [  +30 ms] List of devices attached
[        ] [  +31 ms] executing: /home/mark/Android/Sdk/tools/bin/sdkmanager --licenses
[        ] A crash report has been written to /home/mark/AndroidStudioProjects/lindesk/flutter_02.log.
[        ] This crash may already be reported. Check GitHub for similar crashes.
[   +1 ms] https://github.com/flutter/flutter/issues?q=is%3Aissue+Invalid+argument%28s%29%3A+Cannot+find+executable+for+%2Fhome%2Fmark%2Fdevelopment%2Fflutter%2Fbin%2Fcache%2Fartifacts%2Fengine%2Flinux-x64-release%2Fgen_snapshot.
[        ] To report your crash to the Flutter team, first read the guide to filing a bug.
[        ] https://flutter.dev/docs/resources/bug-reports
[        ] Create a new GitHub issue by pasting this link into your browser and completing the issue template. Thank you!
[        ] https://git.io/Jf6lI
[        ] ninja: build stopped: subcommand failed.
[   +9 ms] Building Linux application... (completed in 5.8s)
[   +1 ms] "flutter linux" took 6,065ms.
Build process failed

#0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
#1      _runBuild (package:flutter_tools/src/linux/build_linux.dart:126:5)
<asynchronous suspension>
#2      buildLinux (package:flutter_tools/src/linux/build_linux.dart:66:11)
<asynchronous suspension>
#3      BuildLinuxCommand.runCommand (package:flutter_tools/src/commands/build_linux.dart:58:11)
#4      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:797:18)
<asynchronous suspension>
#5      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:694:33)
<asynchronous suspension>
#6      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart)
#7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#8      _rootRun (dart:async/zone.dart:1190:13)
#9      _CustomZone.run (dart:async/zone.dart:1093:19)
#10     _runZoned (dart:async/zone.dart:1630:10)
#11     runZoned (dart:async/zone.dart:1550:10)
#12     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#13     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:684:20)
#14     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
#15     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:339:21)
#16     _rootRunUnary (dart:async/zone.dart:1198:47)
#17     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
#18     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
#19     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
#20     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
#21     Future._completeWithValue (dart:async/future_impl.dart:529:5)
#22     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
#23     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
#24     FlutterVersion.checkFlutterVersionFreshness (package:flutter_tools/src/version.dart)
#25     _rootRunUnary (dart:async/zone.dart:1198:47)
#26     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
#27     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
#28     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
#29     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
#30     Future._completeWithValue (dart:async/future_impl.dart:529:5)
#31     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
#32     _rootRun (dart:async/zone.dart:1190:13)
#33     _CustomZone.run (dart:async/zone.dart:1093:19)
#34     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
#35     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
#36     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#37     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#38     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#39     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
```

```
$ flutter analyze
Analyzing lindesk...                                                    
No issues found! (ran in 1.4s)
```

```
$ flutter doctor -v
[✓] Flutter (Channel master, 1.19.0-2.0.pre.213, on Linux, locale en_AU.UTF-8)
    • Flutter version 1.19.0-2.0.pre.213 at /home/mark/development/flutter
    • Framework revision eb1a6efe16 (10 hours ago), 2020-05-30 21:58:02 -0400
    • Engine revision 923687f0e7
    • Dart version 2.9.0 (build 2.9.0-13.0.dev 6489a0c68d)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /home/mark/Android/Sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0
    • cmake version 3.17.2
    • ninja version 1.10.0

[✓] Android Studio (version 3.5)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (1 available)
    • Linux • Linux • linux-x64 • Linux

• No issues found!
```

I am running this on Arch Linux, but using LTS kernel:

```
$ uname -a
Linux ideal 5.4.42-1-lts #1 SMP Wed, 20 May 2020 20:42:53 +0000 x86_64 GNU/Linux
```

--- a/packages/flutter_tools/lib/src/build_system/targets/linux.dart
+++ b/packages/flutter_tools/lib/src/build_system/targets/linux.dart
@@ -23,8 +23,6 @@ const List<String> _kLinuxArtifacts = <String>[
   'flutter_glfw.h',
   // GTK. Not yet used by the template.
   'libflutter_linux_gtk.so',
-  // Shared.
-  'icudtl.dat',
 ];
 
 const String _kLinuxDepfile = 'linux_engine_sources.d';
@@ -86,6 +84,10 @@ class UnpackLinux extends Target {
       outputDirectory: outputDirectory,
       artifacts: _kLinuxArtifacts,
       clientSourcePaths: <String>[clientSourcePath, headersPath],
+      icuDataPath: environment.artifacts.getArtifactPath(
+        Artifact.icuData,
+        platform: TargetPlatform.linux_x64
+      )
     );
     final DepfileService depfileService = DepfileService(
       fileSystem: environment.fileSystem,

--- a/packages/flutter_tools/lib/src/build_system/targets/linux.dart
+++ b/packages/flutter_tools/lib/src/build_system/targets/linux.dart
@@ -86,7 +86,7 @@ class UnpackLinux extends Target {
       clientSourcePaths: <String>[clientSourcePath, headersPath],
       icuDataPath: environment.artifacts.getArtifactPath(
         Artifact.icuData,
-        platform: TargetPlatform.linux_x64
+        platform: TargetPlatform.linux_x64,
       )
     );
     final DepfileService depfileService = DepfileService(

--- a/packages/flutter_tools/lib/src/cache.dart
+++ b/packages/flutter_tools/lib/src/cache.dart
@@ -1392,7 +1392,11 @@ const List<List<String>> _windowsDesktopBinaryDirs = <List<String>>[
 const List<List<String>> _linuxDesktopBinaryDirs = <List<String>>[
   <String>['linux-x64', 'linux-x64/linux-x64-flutter-glfw.zip'],
   <String>['linux-x64', 'linux-x64/flutter-cpp-client-wrapper-glfw.zip'],
+  <String>['linux-x64-profile', 'linux-x64-profile/linux-x64-flutter-glfw.zip'],
+  <String>['linux-x64-release', 'linux-x64-release/linux-x64-flutter-glfw.zip'],
   <String>['linux-x64', 'linux-x64/linux-x64-flutter-gtk.zip'],
+  <String>['linux-x64-profile', 'linux-x64-profile/linux-x64-flutter-gtk.zip'],
+  <String>['linux-x64-release', 'linux-x64-release/linux-x64-flutter-gtk.zip'],
 ];
 
 const List<List<String>> _macOSDesktopBinaryDirs = <List<String>>[

--- a/packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart
+++ b/packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart
@@ -39,7 +39,7 @@ void main() {
     when(mockArtifacts.getArtifactPath(
       Artifact.icuData,
       mode: anyNamed('mode'),
-      platform: anyNamed('platform')
+      platform: anyNamed('platform'),
     )).thenReturn(r'linux-x64/icudtl.dat');
 
     final Environment testEnvironment = Environment.test(

--- a/packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart
+++ b/packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart
@@ -36,6 +36,11 @@ void main() {
       mode: anyNamed('mode'),
       platform: anyNamed('platform'),
     )).thenReturn('linux-x64/flutter_linux');
+    when(mockArtifacts.getArtifactPath(
+      Artifact.icuData,
+      mode: anyNamed('mode'),
+      platform: anyNamed('platform')
+    )).thenReturn(r'linux-x64/icudtl.dat');
 
     final Environment testEnvironment = Environment.test(
       fileSystem.currentDirectory,

--- a/packages/flutter_tools/test/general.shard/cache_test.dart
+++ b/packages/flutter_tools/test/general.shard/cache_test.dart
@@ -611,6 +611,19 @@ void main() {
 
     expect(artifacts.getBinaryDirs(), isNotEmpty);
   });
+
+  testWithoutContext('Linux desktop artifacts include profile and release artifacts', () {
+    final MockCache mockCache = MockCache();
+    final WindowsEngineArtifacts artifacts = LinuxEngineArtifacts(
+      mockCache,
+      platform: FakePlatform(operatingSystem: 'linux'),
+    );
+
+    expect(artifacts.getBinaryDirs(), containsAll(<Matcher>[
+      contains(contains('profile')),
+      contains(contains('release')),
+    ]));
+  });
 }
 
 class FakeCachedArtifact extends EngineCachedArtifact {

--- a/packages/flutter_tools/test/general.shard/cache_test.dart
+++ b/packages/flutter_tools/test/general.shard/cache_test.dart
@@ -614,7 +614,7 @@ void main() {
 
   testWithoutContext('Linux desktop artifacts include profile and release artifacts', () {
     final MockCache mockCache = MockCache();
-    final WindowsEngineArtifacts artifacts = LinuxEngineArtifacts(
+    final LinuxEngineArtifacts artifacts = LinuxEngineArtifacts(
       mockCache,
       platform: FakePlatform(operatingSystem: 'linux'),
     );

